src/journal/journal-file.c

```diff
@@ -3274,8 +3274,6 @@ int journal_file_open(
                         goto fail;
                 }
         } else {
-                assert(fd >= 0);
-
```

Comment:
i'd really keep this in. maybe it's redundant, but all assert()s kinda are. The benefit of keeping this here is that it is "compiled documentation", i.e. it tells the reader that this `else` branch is about the case when fname might not be set but, fd definitely is...

i mean, the function is long, and it doesn't hurt being verbose here, to clarify the context. i mean, we could also add a comment here, but the formalization of this comment as assert() is kinda sexy i think